frontend/lib/src/components/shared/Dropdown/Selectbox.tsx

```diff
@@ -68,9 +67,10 @@ export function fuzzyFilterSelectOptions(
   const filteredOptions = options.filter((opt: SelectOption) =>
     hasMatch(pattern, opt.label)
   )
-  return sortBy(filteredOptions, (opt: SelectOption) =>
-    score(pattern, opt.label)
-  ).reverse()
+  return sortBy(
+    filteredOptions,
+    (opt: SelectOption) => -score(pattern, opt.label, true)
```

Comment:
nit: Could we have a small comment here about why we use `-`? Something like `// we use the negative score here because we want the list to be reversed. We don't reverse it in the end because that can lead to UI shuffling.` (if I understood it correctly from your PR description)